scripts/build.js

```diff
@@ -3,30 +3,144 @@
 const process = require("process");
+const path = require("path");
+
 const cmake = require("cmake-js");
+const axios = require("axios");
+const tar = require('tar');
+
+async function downloadFile(fileUrl, outputLocationPath) {
+    const writer = fs.createWriteStream(outputLocationPath);
+    return axios({
+        method: 'get',
+        url: fileUrl,
+        responseType: 'stream',
+    }).then(response => {
+        return new Promise((resolve, reject) => {
+            response.data.pipe(writer);
+            let error = null;
+            writer.on('error', err => {
+                error = err;
+                writer.close();
+                reject(err);
+            });
+            writer.on('close', () => {
+                if (!error) {
+                    resolve();
+                }
+            });
+        });
+    });
+}
 
-let options = {
-    CMAKE_EXPORT_COMPILE_COMMANDS: true,
-    CMAKE_JS_PLATFORM: os.platform,
-    CMAKE_JS_ARCH: os.arch,
-    BUILD_TESTING: 'OFF',
-    CMAKE_INSTALL_PREFIX: 'crt/install',
-    CMAKE_PREFIX_PATH: 'crt/install',
+async function checkChecksum(url, local_file) {
+    return axios({
+        method: 'get',
+        url: url,
+        responseType: 'text',
+    }).then(response => {
+        return new Promise((resolve, reject) => {
+            const filestream = fs.createReadStream(local_file);
+            const hash = crypto.createHash('sha256');
+            filestream.on('readable', () => {
+                // Only one element is going to be produced by the
+                // hash stream.
+                const data = filestream.read();
+                if (data)
+                    hash.update(data);
+                else {
+                    const checksum = hash.digest("hex")
+                    if (checksum === response.data) {
+                        resolve()
+                    }
+                    else {
+                        reject(new Error("source code checksum mismatch"))
+                    }
+                }
+            });
+        });
+    })
 }
 
-// Convert any -D arguments to this script to cmake -D arguments
-for (const arg of process.argv) {
-    if (arg.startsWith('-D')) {
-        const option = arg.substring(2).split('=')
-        options[option[0]] = option[1]
+async function fetchNativeCode(url, version, path) {
+    const sourceURL = `${url}/aws-crt-${version}-source.tgz`
+    const tarballPath = path + "source.tgz"
+    return new Promise((resolve, reject) => {
+        downloadFile(sourceURL, tarballPath).then(() => {
+            // Download checksum
+            const sourceChecksumURL = `${url}/aws-crt-${version}-source.sha256`
+            checkChecksum(sourceChecksumURL, tarballPath)
+            tar.x({ file: tarballPath, strip: 2, C: nativeSourceDir })
+                .then(() => { resolve(); })
+                .catch((err) => { reject(err); })
+        }).catch((err) => {
+            reject(err)
+        })
+    });
+}
```

Comment:
You async-await instead:
```suggestion
async function fetchNativeCode(url, version, path) {
    const sourceURL = `${url}/aws-crt-${version}-source.tgz`
    const tarballPath = path + "source.tgz";
    await downloadFile(sourceURL, tarballPath);
    const sourceChecksumURL = `${url}/aws-crt-${version}-source.sha256`;
    await checkChecksum(sourceChecksumURL, tarballPath);
    await tar.x({ file: tarballPath, strip: 2, C: nativeSourceDir });
}
```